modules/codegen/src/smithy4s/codegen/internals/SmithyToIR.scala

```diff
@@ -764,6 +751,31 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
     case ConstraintTrait(tr) => Hint.Constraint(toTypeRef(tr), unfoldTrait(tr))
   }
 
+  private def documentationHint(shape: Shape): Option[Hint] = {
+    val memberDocs = shape
+      .members()
+      .asScala
+      .map { member =>
+        val memberDocs = member.getTrait(classOf[DocumentationTrait]).asScala
+        def targetDocs = model
+          .expectShape(member.getTarget)
+          .getTrait(classOf[DocumentationTrait])
+          .asScala
+
+        (
+          member.getMemberName(),
+          memberDocs.orElse(targetDocs)
+        )
+      }
+      .collect { case (name, Some(v)) => (name, v.getValue()) }
+      .toMap
+
+    // todo: what if the shape in question doesn't have docs, but members do?
```

Comment:
I think we'll have to change the `Hint.Documentation` class to account for the shape not having docs. Just having two `Option`s in it might be enough, or a `Ior[ParentDocs, MemberDocs]`